### PR TITLE
Feature/publication

### PR DESCRIPTION
**Pull request by cloedu87**
New Resources and Actions to publish records to a csv files on s3 with or without a custom mapping

- [x] on a `Collection` there is a action to `CollectReviewableRecords` according to the configured `:reviewer` on the collection. As soon as we have the specification on how to collect records for each reviewer, we can implement there a function to filter records according to their needs
- [x] `Export` is the representation of the exported records holding a reference to the exported file and the mapping with which it was created. the Action `publish` creates the file from the records and the mapping on the export and ships it to s3
- [x] statemachine for export
- [x] oban integration to have exports in async jobs
- [x] `runner_test.exs` and `export_test.exs` the test for the new resources

### CODE REVIEW

File: lib/data_aggregator/platform/publication/actions/publish.ex [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+  use Ash.Resource.Actions.Implementation
+
+  alias DataAggregator.DarwinCore.Schema
+  alias DataAggregator.Files.Attachment
+  alias DataAggregator.Platform.Publication.Export
+
+  @impl true
+  def run(input, _opts, _context) do
+    export = DataAggregator.Platform.load!(input.arguments.export, [:records])
```
Comment (hwuethrich) [inline lib/data_aggregator/platform/publication/actions/publish.ex]: This is probably a problem, if there are a lot of records. Maybe you could stream the results? https://hexdocs.pm/ash/Ash.Api.html#c:stream!/2

File: lib/data_aggregator/platform/publication/changes/enque_runner.ex
Comment (hwuethrich) [inline lib/data_aggregator/platform/publication/changes/enque_runner.ex:1]: Typo in file name 🤓

**Review by hwuethrich — CHANGES_REQUESTED** [review]
As discussed :)

File: lib/data_aggregator/platform/publication/actions/publish.ex (hunk 1 above)
Comment (cloedu87) [reply lib/data_aggregator/platform/publication/actions/publish.ex]: done

File: lib/data_aggregator/platform/publication/changes/enque_runner.ex
Comment (cloedu87) [reply lib/data_aggregator/platform/publication/changes/enque_runner.ex:1]: done

File: lib/data_aggregator/records/record/actions/export_records.ex [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    path = "#{Path.join([System.tmp_dir!(), "export"])}#{Ecto.UUID.generate()}.csv"
+
+    attachment =
+      Records.stream!(records_query)
+      |> Stream.map(&map_records(&1, mapping))
+      |> export_to_s3(path, mapping)
+
+    Export.update(export, %{exported_count: Records.count!(records_query)})
+    Export.update_mapping(export, mapping)
+    Export.update_attachment(export, attachment)
```
Comment (hwuethrich) [inline lib/data_aggregator/records/record/actions/export_records.ex]: Maybe add some error handling?

```elixir
with {:ok, export} <- Export.update(export, %{exported_count: Records.count!(records_query)}),
    {:ok, export} <- Export.update_mapping(export, mapping),
    {:ok, export} <- Export.update_attachment(export, attachment) do
  {:ok, export}
end
```

or

```elixir
export
|> Export.update!(%{exported_count: Records.count!(records_query)})
|> Export.update_mapping!(mapping)
|> Export.update_attachment!(attachment)
```

Comment (cloedu87) [reply lib/data_aggregator/records/record/actions/export_records.ex]: thx and check 🙋🏽‍♂️

**Comment by cloedu87** [comment]
it fails because a migration is not committed, but this will be obsolet due to the migrations of the [encoding-pr](https://github.com/zebbra/data_aggregator/pull/92)

**Review by mbaertschi — APPROVED** [review]
Are the headers correctly aligned?

**Review by hwuethrich — APPROVED** [review]

**Review by mbaertschi — APPROVED** [review]

**Pull request merged**